js/private/lifecycle/lifecycle-hooks.js

```diff
@@ -0,0 +1,87 @@
+const {
+    mkdirSync,
+    copyFileSync,
+    readdirSync,
+    readFileSync,
+    existsSync,
+    statSync,
+} = require('fs')
+const path = require('path')
+const { runPostinstallHooks } = require('@pnpm/lifecycle')
+const runLifecycleHook = require('@pnpm/lifecycle').default
+
+async function main(argv) {
+    if (argv.length !== 2) {
+        console.error('Usage: node lifecycle-hooks.js [packageDir] [outputDir]')
+        process.exit(1)
+    }
+    const packageDir = argv[0]
+    const outputDir = argv[1]
+
+    copyPackageContents(packageDir, outputDir)
+
+    const packageJson = JSON.parse(
+        readFileSync(path.join(packageDir, 'package.json'))
+    )
+
+    if (
+        packageJson.scripts?.preinstall ||
+        packageJson.scripts?.install ||
+        packageJson.scripts?.postinstall ||
+        packageJson.scripts?._rules_js_postinstall
+    ) {
+        // The postinstalled copy of the package always one directory beneath the
+        // node_modules folder that npm_import copies into a package's output folder.
+        const nodeModulesPath = path.join(outputDir, '..', 'node_modules')
```

Comment:
perhaps `path.resolve(path.join(outputDir, '..', 'node_modules'))` here so `existsSync` and `mkdirSync` are using the resolve path instead of the path with a `..` segment?